File: Day 29/Solution 1.java
class Solution {
    public int countNodes(TreeNode root) {
        if (root == null) return 0;
        TreeNode left = root.left;
        TreeNode right = root.right;
        int acc = 2;
        while (left != null && right != null) {
            left = left.left;
            right = right.right;
            acc *= 2;
        }
        if (left == right) {
            // in fact it's left = right = null
            return acc - 1;
        } else {
            return 1 + countNodes(root.left) + countNodes(root.right);
    }
}
}

File: Day 76/Solution 1.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def searchBST(self, root, val):
        if not root:
            return []
        
        queue = [root]
        res = []
        
        while queue:
            node = queue.pop(0)
            if val> node.val and node.right:
                queue.append(node.right)
            if val < node.val and node.left:
                queue.append(node.left)
        
            if val == node.val:
                res = node

            
    return res
